code-samples/serving/secrets-go/Dockerfile

```diff
@@ -17,11 +17,22 @@ RUN CGO_ENABLED=0 GOOS=linux GOOS=${TARGETOS} GOARCH=${TARGETARCH} go build -v -
 # https://docs.docker.com/develop/develop-images/multistage-build/#use-multi-stage-builds
 FROM alpine
 
```

Comment:
you probably can do the same as with the Serving images:

- add a go.mod: https://github.com/knative/docs/blob/main/code-samples/serving/multi-container/sidecarcontainer/go.mod
- copy and run go mod: https://github.com/knative/docs/blob/main/code-samples/serving/multi-container/sidecarcontainer/Dockerfile#L14

Reply:
I fix the build error with the following steps:

- `go mod init github.com/knative/docs/code-samples/serving/secrets-go`
- `go mod tidy`
- update Dockerfile as you suggest (copy and run go mod)

Is it okay to upload also the _go.sum_ file?

Reply:
yes, you can add that as well 👍

Reply:
I choose to execute the commands above in the first stage of the Dockerfile, no more build error and the server starts!